app/policies/policy_context.rb

```diff
@@ -0,0 +1,12 @@
+class PolicyContext
+  attr_reader :record
+
+  def initialize(record, user)
+    @record = record
+    @user = user
+  end
+
+  def policy_class
+    "#{@user.roles.uniq.first}DefenceRequestPolicy".classify
```

Comment:
Is roles guaranteed to be ordered? Otherwise this could change from request to request.

Also: should we not have a method on user that will return this in a consistent way? I'm concerned we know too much about the implementation of user, and will have this construct all over our codebase.